lib/jekyll/document.rb

```diff
@@ -123,13 +123,42 @@ def cleaned_relative_path
         relative_path[0..-extname.length - 1].sub(collection.relative_directory, "")
     end
 
+    # Returns an array of sub directories that represents the document's path
+    # relative to the collection directory
+    def subdirs
+      cleaned_relative_path.split(File::SEPARATOR)
+        .reject { |c| c.empty? || c.eql?(basename_without_ext) }
+    end
+
     # Determine whether the document is a YAML file.
     #
     # Returns true if the extname is either .yml or .yaml, false otherwise.
     def yaml_file?
       %w(.yaml .yml).include?(extname)
     end
 
+    # Determine whether the document is a JSON file.
+    #
+    # Returns true if extname == .json, false otherwise.
+    def json_file?
+      extname == ".json"
+    end
+
+    # Determine whether the document is a CSV or TSV file.
+    #
+    # Returns true if extname is either .csv or .tsv, false otherwise.
+    def delimiter_separated_file?
+      %w(.csv .tsv).include?(extname)
+    end
+
+    # Determine the column delimiter for ___SeparatedValue files.
+    #
+    # Returns "\t" if extname is `.tsv`, otherwise returns ",".
+    def column_delimiter
+      return "\t" if extname == ".tsv"
+      ","
```

Comment:
Fine, we can deal with this in a dedicated PR.

Reply:
Yea, I figured no need to add a configuration option until someone reports an issue as it really is a rare thing. I figured it might be good to point out why I decided on this code pattern for a return.